### ISSUE TITLE
Function returning generic `Callable` unchecked

### ISSUE THREAD

**Issue by kwshi** [microsoft/pyright#2770]
**Describe the bug**
If a parametrized function returns a `typing.Callable` with generic argument/return types, the returned callable's arg/ret types are unchecked.

**To Reproduce**
Here are some examples, showing unsound code passing type-checks.
```python
import typing as t
T = t.TypeVar("T")


# generic argument type unchecked
def berries(_: T) -> t.Callable[[T], None]:
    def f(a: str):
        a.split()
    return f # should fail because `f` is monomorphic `t.Callable[[str], None]`

# unsound usage example
b = berries(3) # inferred type `t.Callable[[int], None]`
b(15) # tries to invoke `15.split()`, type error


# generic return types unchecked
def asparagus(_: T) -> t.Callable[[], T]:
    def f():
        return "hello"
    return f # should fail because `f` always returns `str`, not `T`

def melons(_: T) -> t.Callable[[], T]:
    return lambda: "hello"

def potatoes(_: T) -> t.Callable[[], T]:
    f = lambda: "hello"
    return f

def grass(_: list[T]) -> t.Callable[[], T]:
    return lambda: "hello"

# unsound usage
y: list[bool] = []
g = grass(y) # inferred type `t.Callable[[], bool]`
x = g() # inferred type `bool`, actually returns `"hello"::str`
```
All of these currently pass type-checking.

**Expected behavior**
These functions should all fail to type-check.

**VS Code extension or command-line**
command-line language server via neovim LSP config; pyright 1.1.201

**Comment by erictraut** [comment]
Thanks for the bug report. I appreciate the detailed repro case!

This will be fixed in the next release of pyright.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.202, which I just published. It will also be included in the next release of pylance.
